llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -567,6 +567,19 @@ void llvm::applyGlobalValOffset(MachineInstr &MI, MachineRegisterInfo &MRI,
       B.buildConstant(LLT::scalar(20), -static_cast<int64_t>(Offset)));
 }
 
+/// Check if the Intrinsic can produce S20
+static bool isS20Intrinsic(const MachineInstr &MI,
+                           const MachineRegisterInfo &MRI) {
+  const unsigned IntrinsicID = cast<GIntrinsic>(MI).getIntrinsicID();
+  if (IntrinsicID == Intrinsic::aie2_vextract_elem8_I512 ||
+      IntrinsicID == Intrinsic::aie2_vextract_elem16_I512 ||
+      IntrinsicID == Intrinsic::aie2_vextract_elem32_I512) {
+    // Check if the sign value is constant
+    return getIConstantVRegSExtVal(MI.getOperand(4).getReg(), MRI).has_value();
```

Comment:
define a new const Reg with a meaning full name. I guess `index` will be a good candidate.